src/Npgsql/Internal/TypeHandlers/TextHandler.cs

```diff
@@ -270,5 +271,13 @@ public Task Write(byte[] value, NpgsqlWriteBuffer buf, NpgsqlLengthCache? length
     #endregion
 
     /// <inheritdoc />
-    public virtual TextReader GetTextReader(Stream stream) => new StreamReader(stream, _encoding);
+    public virtual TextReader GetTextReader(Stream stream, int byteLength, NpgsqlReadBuffer buffer)
+    {
+        if (byteLength >= 0 && buffer.ReadBytesLeft >= byteLength)
```

Comment:
Is there a situation where byteLength < 0 will occur? If not, better assert against it no?

Reply:
> Weird, in ReplicationValue we call GetTextReader, passing it the result of GetStream(). That function always returns a ColumnStream with a specific byte Length, so we do know the length and should be able to pass it to the text reader as well... What do you think?

Honestly, I haven't really looked there at all :trollface: But yeah, there is a length property.

I think I'm going to follow @Brar's suggestion and just use the remaining length of `ColumnStream` (`Length` - `Position`). One last review...